pageserver/src/tenant/ephemeral_file/page_caching.rs

```diff
@@ -49,6 +58,46 @@ impl RW {
         self.rw.bytes_written()
     }
 
+    /// Load all blocks that can be read via [`Self::read_blk`] into a contiguous memory buffer.
+    ///
+    /// This includes the blocks that aren't yet flushed to disk by the internal buffered writer.
+    /// The last block is zero-padded to [`PAGE_SZ`], so, the returned buffer is always a multiple of [`PAGE_SZ`].
+    pub(super) async fn load_into_contiguous_memory(
```

Comment:
There's potential for confusion in this name: sometimes when an I/O function talks about contiguous memory it means physically contiguous.  Perhaps just `load_to_vec`?